docs/interfaces.md

```diff
@@ -12,198 +12,362 @@ Slots may support multiple connections to plugs.  For example the OS snap
+    $ snap interfaces
+    Slot                 Plug
+    :log-observe         foo:log-observe
 
-Can access the PulseAudio sound server. Allows for sound playback in games and
-media application. It doesn't allow recording.
+and disconnect using ``snappy disconnect``:
 
-Usage: common
-Auto-Connect: yes
+    $ sudo snap disconnect foo:log-observe core:log-observe
+    $ snap interfaces # shows they are disconnected
+    Slot                 Plug
+    :log-observe         -
+    -                    foo:log-observe
 
-### opengl
+On the other hand, ``bar`` could use ``plugs: [ network ]`` and since
+``network`` is auto-connected, ``bar`` has access to the interface upon
+install:
 
-Can access the opengl hardware.
+    $ sudo snap install bar
+    $ snap interfaces
+    Slot                 Plug
+    :home                bar:home
 
-Usage: reserved
-Auto-Connect: yes
+You may disconnect an auto-connected interface:
 
-### home
+    $ sudo snap disconnect bar:home core:home
+    $ snap interfaces
+    Slot                 Plug
+    :home                -
+    -                    bar:home
+
+Whether the slot is provided by the OS snap or not doesn't matter in terms of
+snap interfaces except that if the slot is provided by a snap, a snap that
+implements the slot must be installed for it to be connectable. Eg, the
+``bluez`` interface is not provided by the OS snap so a snap author
+implementing the bluez service might use ``slots: [ bluez ]``. Then after
+install, the bluez interface shows up as available:
+
+    $ sudo snap install foo-blue
+    $ snap interfaces
+    Slot                 Plug
+    foo-blue:bluez       -
 
-Can access non-hidden files in user's `$HOME` to read/write/lock.
-This is restricted because it gives file access to the user's
-`$HOME`. This interface is auto-connected on classic systems and
-manually connected on non-classic.
+Now install and connect works like before (eg, ``baz`` uses
+``plugs: [ bluez ]``):
 
-Usage: reserved
-Auto-Connect: yes
+    $ sudo snap install baz
+    $ snap interfaces
+    Slot                 Plug
+    foo-blue:bluez       -
+    -                    baz:bluez
+    $ sudo snap connect baz:bluez foo-blue:bluez
+    $ snap interfaces
+    Slot                 Plug
+    foo-blue:bluez       baz:bluez
+
+## Supported Interfaces - Basic
+
+### camera
+
+Can access the first video camera. Suitable for programs wanting to use
+webcams.
+
+* Auto-Connect: no
+* Availability: OS snap (classic)
 
 ### gsettings
 
-Can access global gsettings of the user's session. This is restricted because
-it gives privileged access to sensitive information stored in gsettings and
-allows adjusting settings of other applications.
+Can access global gsettings of the user's session which gives privileged access
+to sensitive information stored in gsettings and allows adjusting settings of
+other applications.
 
-Usage: reserved
-Auto-Connect: yes
+* Auto-Connect: yes
+* Availability: OS snap (classic)
```

Comment:
These should all be "core snap" now, but I'm not quite sure we should be documenting that as such. Devices may easily end up not offering certain slots, and other slots may easily move around to different snaps. The point of our interface design is precisely that we're not making any promises about what is available where, yet the system will behave correctly independently of that.